packages/protocol/engine/tangle/blockdag/blockdag.go

```diff
@@ -130,6 +130,10 @@ func (b *BlockDAG) evictEpoch(epochIndex epoch.Index) {
 }
 
 func (b *BlockDAG) markSolid(block *Block) (err error) {
+	if err = b.checkTimestampMonotonicity(block); err != nil {
```

Comment:
This is executed as part of the causalorder? meaning if it returns an error, the block is marked as invalid right?

Reply:
Exactly.